distros/catkin-setup-hook/setup-hook.sh

```diff
@@ -76,6 +76,13 @@ _catkinPreConfigureHook() {
 }
 preConfigureHooks+=(_catkinPreConfigureHook)
 
+_exportCMakePrefixPath() {
+  # Needed for https://github.com/NixOS/nixpkgs/commit/8c9c8ade2f88a85ccdd4858cc802d7b7d6c48fe0
+  # see: https://github.com/lopsided98/nix-ros-overlay/issues/491
+  export CMAKE_PREFIX_PATH=$NIXPKGS_CMAKE_PREFIX_PATH
```

Comment:
I'd just add this to `_catkinPreConfigureHook`. It's not necessary to create a new function.

And the comment could mention why it's needed - I propose something like this:

```
Catkin uses CMAKE_PREFIX_PATH to find additional workspaces, but
nixpkgs switched to using NIXPKGS_CMAKE_PREFIX_PATH
(https://github.com/NixOS/nixpkgs/commit/8c9c8ade2f88a85ccdd4858cc802d7b7d6c48fe0).
see: https://github.com/lopsided98/nix-ros-overlay/issues/491
```

I also wonder whether it's necessary to export the variable to all processes running during build. It seems it should be sufficient to pass it just to cmake via command line switches:

```sh
cmakeFlags+=" -DCMAKE_PREFIX_PATH=\"$NIXPKGS_CMAKE_PREFIX_PATH\""
```

Can you check, whether this works? If yes, I'd prefer this variant, because it might help with some cross-compilation scenarios, which are fixed by the nixpkgs commit.